### ISSUE TITLE
Crash on Android Error: java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by hammadbawara** [bluefireteam/audioplayers#1274]
## Checklist
- [x] I read the [troubleshooting guide](https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md) before raising this issue
- [x] I made sure that the issue I am raising doesn't already exist

## Current bug behaviour

App crahses when user press many buttons at once in Xylophone app.

## Expected behaviour

I made a Xylophone App on flutter. You can click different buttons and it play different audios. But when use press many buttons in fast speed this app crashes and show this error. I only tested this app on Android.

## Steps to reproduce

```
void playNote(int noteNumber){
    final player = AudioPlayer();
    player.play(AssetSource('note$noteNumber.wav'));
  }
```

<details>
<summary>Code sample</summary>

```dart
void main() {
}
```

</details>

## Logs

```
E/AndroidRuntime( 6063): java.util.ConcurrentModificationException
E/AndroidRuntime( 6063): 	at java.util.LinkedHashMap$LinkedHashIterator.nextEntry(LinkedHashMap.java:411)
E/AndroidRuntime( 6063): 	at java.util.LinkedHashMap$ValueIterator.next(LinkedHashMap.java:426)
E/AndroidRuntime( 6063): 	at xyz.luan.audioplayers.AudioplayersPlugin$UpdateCallback.run(AudioplayersPlugin.kt:208)
E/AndroidRuntime( 6063): 	at android.os.Handler.handleCallback(Handler.java:751)
E/AndroidRuntime( 6063): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime( 6063): 	at android.os.Looper.loop(Looper.java:154)
E/AndroidRuntime( 6063): 	at android.app.ActivityThread.main(ActivityThread.java:6077)
E/AndroidRuntime( 6063): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 6063): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
E/AndroidRuntime( 6063): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
```

<details>
  <summary>Full Logs</summary>

  

  ```
  my full logs or a link to a gist
  ```

  Flutter doctor:
  ```
  Output of: flutter doctor -v
  ```
</details>

## Audio Files/URLs/Sources

- Audio Files : https://drive.google.com/drive/folders/1Hf5sWMtroSgYJGV3NSlLQCwO6Q0EDbES?usp=sharing

## Environment information

* audioplayers version: 1.0.1

Platform 1: 
* OS name and version:  Android 9 (Physical), Android 6 (Emulator)
* Device: Mobile

* build mode: `debug`

* error peculiarities (optional): 

Platforms tested without any issue (optional): 
* test peculiarities:

**Comment by Gustl22** [comment]
You can also test now with the newest release `audioplayers: ^1.1.0`

**Comment by Sherif-2001** [comment]
I have the same issue; I am using the latest version of audioplayers and Flame_Audio and this error appears when I play a bgm with sound effects

**Comment by spydon** [comment]
@luanpotter @Gustl22 any clue how this could be solved?
I'm not proficient enough with how the map is passed between Dart and Kotlin, it seems like it's not possible to pass a `ConcurrentMap` at least and I've also tried to make a copy of the map before looping it but without success, it still throws the `ConcurrentModificationException`.
We have a lot of people reporting this error on Discord.

**Comment by spydon** [comment]
I just released v1.1.1 with a fix for this, @Gustl22 saves the day again!
![image](https://user-images.githubusercontent.com/744771/194716057-3d124327-c4bf-47f3-943c-72e2948efe56.png)
